code-challenges.js

```diff
@@ -7,7 +7,32 @@ var secretCodeWord1 = "lackadaisical"
 var secretCodeWord2 = "gobbledygook"
 // Expected output: "g0bbl3dyg00k"
 
+// declare function called secretWord that takes argument of string
+const secretWord = (string) => {
+    // convert string to array
+    // map through the array and check each letter and its index
+    let stringToArr = string.split("")
+    return stringToArr.map(letter => {
+        // make sure all characters are lowercase for check
+        let lowerCasedLetter = letter.toLowerCase()
+        // reassign letters based on conditions:
+        // if letter is a, assign to be 4; if e, assign to 3; if i, assign to 1; if o, assign to 0
+        // try ternary operator instead of if..else if..else?
+        return lowerCasedLetter === 'a'? letter = '4'
+        : lowerCasedLetter === 'e'? letter = '3'
+        : lowerCasedLetter === 'i'? letter = '1'
+        :lowerCasedLetter === 'o'? letter = '0'
+        // else if none of the above, just return the letter as is
+        : letter
+        // chain the join method to convert array back to string
+    }).join("")
+    
+}
```

Comment:
I love this solution! love the use of a ternary instead of if/else! very cool ;)